### ISSUE TITLE
Intake v2: investigate parallelizing decode and validate

### ISSUE THREAD

**Issue by roncohen** [elastic/apm-server#1285]
As [previously discussed](https://github.com/elastic/apm-server/pull/1260#pullrequestreview-145137222), it would be interesting to investigate parallelizing the decoding and validation of documents. Transformation is already being done in parallel.

At the moment, decoding and validation is happens in parallel at the request level. E.g. incoming data from separate requests are being decoded and validated in parallel. As part of this issue, we should also seek to put a bound on that parallelization

**Comment by alvarolobato** [comment]
We agreed that it's a bit premature to implement this know, not even needed probably. Deferred from 6.5

**Comment by graphaelli** [comment]
@axw do the changes you've proposed and implemented making this obsolete?

**Comment by axw** [comment]
Nope.

https://github.com/elastic/apm-server/issues/3551 will weave validation and decoding together such that parallelising them wouldn't make sense.

It might make sense to decode concurrently, e.g. have the stream processor send events to a channel to be decoded, and then have multiple goroutines processing those. This would enable parallel decoding within a single stream.

**Comment by stuartnelson3** [comment]
Closing this for now as we aren't prioritizing time to investigate whether it's worth implementing this
